models/spike_detector.cpp

```diff
@@ -82,37 +78,56 @@ nest::spike_detector::init_buffers_()
 void
 nest::spike_detector::calibrate()
 {
+
   if ( kernel().event_delivery_manager.get_off_grid_communication() )
   {
-    // If precise models exist, it is necessary to ensure that precise_times
-    // property
-    // is set to true and the precision should increase from the default value
-    // of 3.
-
-    // Default precision value for simulations involving precise models.
-    long precision = 15;
 
-    if ( user_set_precision_value != -1 )
+    if ( !device_.is_precise_times_user_set() )
     {
-      // If precision is set to a value by the user, it must be used instead of
-      // the default.
-      precision = user_set_precision_value;
+
+      if ( !device_.is_precision_user_set() )
+      {
+        // it makes sense to increase the precision if precise models are used.
+        device_.set_precise(true, 15);
+        LOG( M_INFO,
+             "spike_detector::calibrate",
+             String::compose(
+                     "Precise neuron models exist: the property precise_times "
+                             "of the %1 with gid %2 has been set to true, precision has "
+                             "been set to 15.",
+                     get_name(),
+                     get_gid() ) );
+      }
+
+      else
+      {
+        device_.set_precise(true, device_.get_precision());
```

Comment:
Yes, this is a bit confusing. I want to change set_precise method later and add two separate methods for changing precision and precise_times in a later version. If I remove this line, if precise models exist and user has not set "precise_times", it would not be set to true.